### ISSUE TITLE
I think adding new function calling api would make gpt-engineer more robust

### ISSUE THREAD

**Issue by ezzye** [AntonOsika/gpt-engineer#192]
see https://openai.com/blog/function-calling-and-other-api-updates
and https://platform.openai.com/docs/guides/gpt/function-calling
and https://youtu.be/0lOSvOoF2to

**Comment by patillacode** [comment]
Yes, there are things on track with functions, thanks for the recommendation.
